src/hooks/useTransactionSteps.ts

```diff
@@ -121,42 +117,43 @@ const useTransactionSteps = ({ transaction, onComplete }: Props) => {
       {
         chainType: targetChain,
         label: 'Deliver message',
-        labelOnChain: onChainCompleted(messageDelivered) && transaction.messageNonce,
+        labelOnChain: onChainCompleted(messageDelivered) && deliveryBlock,
         status: completionStatus(messageDelivered)
       },
       {
         chainType: targetChain,
         label: 'Finalize message in target block',
-        labelOnChain: onChainCompleted(messageFinalizedOnTarget) && targetMessageDeliveryBlock,
+        // TODO: To apply a different design for blocks and nonces: #175
+        labelOnChain: onChainCompleted(messageFinalizedOnTarget) && deliveryBlock,
         status: completionStatus(messageFinalizedOnTarget)
       },
       {
         chainType: sourceChain,
         label: 'Confirm delivery',
-        status: completionStatus(sourceConfirmationReceived)
+        status: completionStatus(messageFinalizedOnTarget && sourceConfirmationReceived)
```

Comment:
Why is this extra check needed? Like it is not possible to have confirmation before finality.